PR: https://github.com/odoo/odoo/pull/178200

From: f10bd8ddcf99c9247dca417fefbc260454d74ea1
From: Gorash
Date: 2024-10-15 09:33:02

Breaking data model changes scores: del:8 + add:0, change matches:
-    _inherit = "ir.model"
-    _inherit = 'ir.qweb.field.image'
-    _inherit = "ir.ui.menu"
-    _inherit = 'ir.ui.view'
-    _inherit = 'res.company'
-    _inherit = 'res.config.settings'
-    _inherit = 'res.partner'
-    _inherit = "res.users"

Total Changes: 34

[IMP] *: adapt model class names to correspond to model names (apply script)

Part-of: odoo/odoo#178200
Related: odoo/enterprise#69762
Related: odoo/documentation#11115
Related: odoo/upgrade#6577
Related: odoo/design-themes#988
Related: odoo/upgrade-util#143

================================= pseudo patch: =================================

--- a/addons/web/models/base_document_layout.py
+++ b/addons/web/models/base_document_layout.py
@@ -27,7 +27,6 @@ class BaseDocumentLayout(models.TransientModel):
     Customise the company document layout and display a live preview
     """
 
-    _name = 'base.document.layout'
     _description = 'Company Document Layout'
 
     @api.model

--- a/addons/web/models/ir_http.py
+++ b/addons/web/models/ir_http.py
@@ -27,8 +27,8 @@ comma (eg: 'tests, assets').
 ALLOWED_DEBUG_MODES = ['', '1', 'assets', 'tests', 'disable-t-cache']
 
 
-class Http(models.AbstractModel):
-    _inherit = 'ir.http'
+class IrHttp(models.AbstractModel):
+    _inherit = ['ir.http']
 
     bots = ["bot", "crawl", "slurp", "spider", "curl", "wget", "facebookexternalhit", "whatsapp", "trendsmapresolver", "pinterest", "instagram"]
 

--- a/addons/web/models/ir_model.py
+++ b/addons/web/models/ir_model.py
@@ -5,7 +5,7 @@ from odoo import models, api
 
 
 class IrModel(models.Model):
-    _inherit = "ir.model"
+    _inherit = ["ir.model"]
 
     @api.model
     def display_name_for(self, models):

--- a/addons/web/models/ir_qweb_fields.py
+++ b/addons/web/models/ir_qweb_fields.py
@@ -10,16 +10,15 @@ from odoo import api, models, fields
 from odoo.tools import html_escape as escape
 
 
-class Image(models.AbstractModel):
+class IrQwebFieldImage(models.AbstractModel):
     """
     Widget options:
 
     ``class``
         set as attribute on the generated <img> tag
     """
-    _name = 'ir.qweb.field.image'
     _description = 'Qweb Field Image'
-    _inherit = 'ir.qweb.field.image'
+    _inherit = ['ir.qweb.field.image']
 
     def _get_src_urls(self, record, field_name, options):
         """Considering the rendering options, returns the src and data-zoom-image urls.
@@ -113,9 +112,10 @@ class Image(models.AbstractModel):
 
         return Markup(''.join(img))
 
-class ImageUrlConverter(models.AbstractModel):
+
+class IrQwebFieldImage_Url(models.AbstractModel):
     _description = 'Qweb Field Image'
-    _inherit = 'ir.qweb.field.image_url'
+    _inherit = ['ir.qweb.field.image_url']
 
     def _get_src_urls(self, record, field_name, options):
         image_url = record[options.get('preview_image', field_name)]

--- a/addons/web/models/ir_ui_menu.py
+++ b/addons/web/models/ir_ui_menu.py
@@ -7,7 +7,7 @@ from odoo import models
 
 
 class IrUiMenu(models.Model):
-    _inherit = "ir.ui.menu"
+    _inherit = ["ir.ui.menu"]
 
     def load_web_menus(self, debug):
         """ Loads all menu items (all applications and their sub-menus) and

--- a/addons/web/models/ir_ui_view.py
+++ b/addons/web/models/ir_ui_view.py
@@ -3,8 +3,8 @@
 from odoo import models
 
 
-class View(models.Model):
-    _inherit = 'ir.ui.view'
+class IrUiView(models.Model):
+    _inherit = ['ir.ui.view']
 
     def get_view_info(self):
         _view_info = self._get_view_info()

--- a/addons/web/models/models.py
+++ b/addons/web/models/models.py
@@ -37,7 +37,7 @@ DISPLAY_DATE_FORMATS = {
 
 
 class Base(models.AbstractModel):
-    _inherit = 'base'
+    _inherit = ['base']
 
     @api.model
     @api.readonly
@@ -1049,9 +1049,8 @@ class Base(models.AbstractModel):
                 self.update_field_translations(field_name, translations)
 
 
-
 class ResCompany(models.Model):
-    _inherit = 'res.company'
+    _inherit = ['res.company']
 
     @api.model_create_multi
     def create(self, vals_list):

--- a/addons/web/models/res_config_settings.py
+++ b/addons/web/models/res_config_settings.py
@@ -5,6 +5,6 @@ from odoo import fields, models
 
 
 class ResConfigSettings(models.TransientModel):
-    _inherit = 'res.config.settings'
+    _inherit = ['res.config.settings']
 
     web_app_name = fields.Char('Web App Name', config_parameter='web.web_app_name')

--- a/addons/web/models/res_partner.py
+++ b/addons/web/models/res_partner.py
@@ -16,7 +16,7 @@ except ImportError:
 
 
 class ResPartner(models.Model):
-    _inherit = 'res.partner'
+    _inherit = ['res.partner']
 
     def _build_vcard(self):
         """ Build the partner's vCard.

--- a/addons/web/models/res_users.py
+++ b/addons/web/models/res_users.py
@@ -5,7 +5,7 @@ from odoo.osv import expression
 
 
 class ResUsers(models.Model):
-    _inherit = "res.users"
+    _inherit = ["res.users"]
 
     @api.model
     def name_search(self, name='', args=None, operator='ilike', limit=100):
